### PR TITLE
SilverStripe 4 and Namespacing

### PR DESCRIPTION
**Pull request by Firesphere**

### CODE REVIEW

**Comment by tractorcow** [comment]
Nice job, just need to set CORE_RELEASE=master. :)

**Comment by tractorcow** [comment]
When you get tests to pass I'll branch master into new major version.

Can you alias dev-master as 3.0.x-dev in composer.json?

**Comment by Firesphere** [comment]
Yeah, I seem to have issues getting the tests to run on master. I need to figure out what I'm doing wrong :|

**Comment by tractorcow** [comment]
You need to adjust CORE_RELEASE in `.travis.yml` and also remove any php versions below 5.5, which is the min version for 4.0.

**Comment by robbieaverill** [comment]
>You need to adjust CORE_RELEASE in .travis.yml and also remove any php versions below 5.5, which is the min version for 4.0.

It's now 5.6 :D

**Comment by robbieaverill** [comment]
I've rebased this and updated the PR with namespaces and some other things

**Comment by robbieaverill** [comment]
I'm just in the process of merging up the new changes from 2.x, so will rebase again once that's done

**Comment by robbieaverill** [comment]
Please note: I haven't added a legacy remapping definition for `EditableSpamProtectionField`, because if the class doesn't exist then dev/build throws an exception. This class is only defines at runtime with [SS4 compatible] userforms installed as well, so it doesn't seem appropriate. I wonder whether we could move this form field into a new module like `silverstripe/userforms-spamprotection` or something to make this a bit cleaner?

**Comment by dhensby** [comment]
> Please note: I haven't added a legacy remapping definition for EditableSpamProtectionField, because if the class doesn't exist then dev/build throws an exception. This class is only defines at runtime with [SS4 compatible] userforms installed as well, so it doesn't seem appropriate. I wonder whether we could move this form field into a new module like silverstripe/userforms-spamprotection or something to make this a bit cleaner?

I wouldn't object to pulling that dependency out and adding it as a module.

File: README.md [hunk 1]
```diff
@@ -33,95 +32,113 @@ implements the `SpamProtector` interface. More on that below.
...
-The logic to perform the actual spam validation is controlled by each of the 
-individual `SpamProtector` implementation since they each require a different 
+```php
+// your existing form code
+$form = new Form(/* .. */);
+$form->enableSpamProtection();
+```
+
+The logic to perform the actual spam validation is controlled by each of the
+individual `SpamProtector` implementation since they each require a different
```
Comment (fspringveldt) [inline README.md]: [Minor] Consider rewording to either
_The logic to perform the actual spam validation is controlled by each individual `SpamProtector` implementation ..._
or
_The logic to perform the actual spam validation is controlled by each of the individual `SpamProtector` implementations ... _

**Pull request merged**